### DISCUSSION TITLE
Make localized URLs shareable between users with different locales

### DISCUSSION THREAD

**Discussion by marvin-wtt** [spatie/laravel-sluggable#228] [Ideas]
## Why?

With the current implementation of implicit route model binding, it is only possible to query for slugs within the same locale. This becomes a problem when a link with a localized slug is shared to a person with a different locale configured. Even though the model exists, the user will get a 404.  

Before creating a PR to address this issue, I'd like to hear a few opinions from you and @freekmurze on my thoughts so far. 
I basically have two approaches, which are explained here.
I'd of course make this optional to avoid a BC.

## Option 1

Search for the slug with [REGEXP](https://dev.mysql.com/doc/refman/8.0/en/regexp.html#operator_regexp)
This is by far the easiest approach, but noticeably worse performance wise for larger datasets.

## Option 2

Create a separate table for the translated slugs which morphs to the model instead of storing them as JSON. This would be much better than executing a regex search on every page request.  The slugs would then need to be unique for `[sluggable_type, name]`.
The user experience would be similar IMO. It would only be required to publish the migration if the feature is wanted.

The only downside I could imagine is that it is slightly more complicated for the user to manually override a slug, as some admin panels already provide great support for laravel-translateable fields.

```
slugs
    id - integer
    sluggable_type - string
    sluggable_id - integer
    locale - string
    name - string
```

**Comment by vkartsivadze** [comment]
Is there any update regarding this?

**Reply by vkartsivadze** [reply]
My solution so far

``` php
public function show($slug)
    {
        $blogPost = BlogPost::findBySlug($slug);

        if (! $blogPost) {
            foreach (config('app.locales') as $locale) {
                $blogPost = BlogPost::where('slug->'.$locale, $slug)->first();
                if ($blogPost) {
                    break;
                }
            }
        }

        abort_unless($blogPost,404);

        return view('blog.show', [
            'blogPost' => $blogPost,
        ]);
    }

**Reply by vkartsivadze** [reply]
Improved version, it will redirect to localized slug instead of showing original slug but correct content

```php
    public function show($slug)
    {
        $article = Article::findBySlug($slug);

        if (! $article) {
            foreach (config('app.locales') as $locale) {
                $article = Article::where('slug->'.$locale, $slug)->first();
                if ($article) {
                    return redirect()->to(route('articles.show', ['article' => $article]));
                }
            }
        }

        abort_unless($article,404);

        return view('articles.show', compact('article'));
    }
```

**Reply by marvin-wtt** [reply]
I created a PR but it was not merged.

See:
https://github.com/spatie/laravel-sluggable/pull/233

**Reply by vkartsivadze** [reply]
It is a must have feature with translatable slugs :/

**Reply by marvin-wtt** [reply]
You can tag Freek in the PR or here in the discussion. Maybe he changed his opinion on that subject
